src/Powercord/managers/plugins.js

```diff
@@ -56,34 +48,9 @@ module.exports = class PluginManager {
     return !powercord.settings.get('disabledPlugins', []).includes(plugin);
   }
 
-  isEnforced (plugin, iterate = true) {
-    if (this.enforcedPlugins.includes(plugin)) {
-      return true;
-    }
-
-    if (!iterate) {
-      return false;
-    }
-
-    const dependents = this.resolveDependents(plugin);
-    return dependents.filter(p => this.isEnforced(p, false)).length !== 0;
-  }
-
   // Resolvers
-  async resolveDependencies (plugin, deps = []) {
-    const dependencies = await this.getDependencies(plugin);
-
-    await Promise.all(dependencies.map(async dep => {
-      if (!deps.includes(dep)) {
-        deps.push(dep);
-        deps.push(...(await this.resolveDependencies(dep, deps)));
-      }
-    }));
-    return deps.filter((d, p) => deps.indexOf(d) === p);
-  }
-
   resolveDependents (plugin, dept = []) {
-    const dependents = this.getAllPlugins().filter(p => this.getDependenciesSync(p).includes(plugin));
+    const dependents = this.getPlugins().filter(p => this.getDependenciesSync(p).includes(plugin));
```

Comment:
Change the callback function found under 'filter' to `this.get(p).allDependencies.includes(plugin)` as the `getDependenciesSync(p)` method was removed (since 70c1c05) and is now deprecated.

**Note**: Leaving this line untouched prevents plug-ins from being disabled (or uninstalled) straight from the 'Plugins' tab of user settings.

Reply:
I'm aware, but keep in mind that the plugin manager have never been here to work, that's why there is a red banner :)